### PR TITLE
run e2e tests on CI

### PR DESCRIPTION
**Pull request by mathieudutour**
I needed to rework a bit some part because when running the tests, we only get an `MSDocumentData` instead of an `MSDocument`. That's actually fine because we only need that. Just need to make sure that we use the documentData instead of the document throughout the lib.

### CODE REVIEW

**Comment by ljharb** [comment]
What only supports node 8, and why?

**Comment by mathieudutour** [comment]
`@skpm/test-runner` only supports node 8+ and because it's using async/await and I don't want to compile it

**Comment by ljharb** [comment]
Why not transpile it, or write in normal Promises instead of using the sugar?

**Comment by mathieudutour** [comment]
I'm not sure why we are discussing it here. It's a test runner that only supports node 8+ that's about it. I'm not even sure why we are running the tests on different node versions since this library is compiled anyway...

**Comment by ljharb** [comment]
The library being transpiled is irrelevant; it still has to be verified to run on supported node versions.

If a test runner won't work on an LTS version of node, then I don't think it's really usable on node projects.

**Comment by mathieudutour** [comment]
>  it still has to be verified to run on supported node versions

This is not a node library so there are no supported node versions? What we could (and maybe should) test is different macOS versions and different Sketch versions.

> If a test runner won't work on an LTS version of node, then I don't think it's really usable on node projects

Again, that's not a node project.

I'm really not sure why we are discussing this. If you don't want e2e testing then fine, let's just drop it /shrug

**Comment by ljharb** [comment]
I think e2e testing is great, and I don't want to drop it.

I was reacting to the comment about the tests failing on node 6.

If this isn't a node project, then that would mean node isn't required to be installed to use it - that doesn't seem like the case tho.

**Comment by mathieudutour** [comment]
the only thing node is used for is installing the library. It doesn't run in node in any way.

**Comment by ljharb** [comment]
(assuming you mean the whole project; not just the test runner) In that case, perhaps there's no value in running the tests on multiple node versions at all, and we can replace all three of 6/8/10 with `lts/*`?

File: __tests__/skpm/basic.test.js [hunk 1]
```diff
@@ -2,6 +2,10 @@ import * as React from 'react';
 import * as sketch from 'sketch'; // eslint-disable-line
 import { render, View, Artboard, Text } from '../../src';
 
+function getDoc(context, document) {
```
Comment (jaridmargolin) [inline __tests__/skpm/basic.test.js]: Could you add a comment regarding what this function is doing, and why it is necessary?

File: src/symbol.js [hunk 2]
```diff
@@ -32,13 +33,13 @@ const msListToArray = (pageList) => {
   return out;
 };
 
-export const getSymbolsPage = (document: any) => {
+export const getSymbolsPage = (document: SketchDocumentData) => {
```
Comment (jaridmargolin) [inline src/symbol.js]: 🌈

File: src/symbol.js [hunk 3]
```diff
@@ -32,13 +33,13 @@ const msListToArray = (pageList) => {
...
   const pages = document.pages();
   const array = msListToArray(pages);
   return array.find(p => String(p.name()) === 'Symbols');
 };
 
-const getExistingSymbols = (document: any) => {
+const getExistingSymbols = (document: SketchDocumentData) => {
```
Comment (jaridmargolin) [inline src/symbol.js]: ✨

**Comment by mathieudutour** [comment]
> Could you provide a little insight into why everything exists on document.documentData() rather than document?

`document` is an instance of `MSDocument` which is a subclass of `NSDocument`.
`document.documentData()` is an instance of `MSDocumentData` which is the root of our data model.

All the manipulations of the model happens on the MSDocumentData but there are some convenience methods on MSDocument that just forward it to the MSDocumentData.

For `react-sketchapp`, we only want to deal with MSDocumentData but we can receive an MSDocument instance so we have to make sure we access its document data if that's the case.

**Comment by jaridmargolin** [comment]
Would it then make sense to rename occurrences of `document` -> `documentData`? Likewise, would it make sense to rename the helper methods:

`getDocumentFromContainer` -> `getDocumentDataFromContainer`
`getDocumentFromContext` -> `getDocumentDataFromContext`

Definitely a little verbose, but removes any possible confusion.

**Pull request merged**